File: android/app/src/main/java/de/repeatuntil/yata/domain/todolist/TodoListRepository.kt
package de.repeatuntil.yata.domain.todolist

import de.repeatuntil.yata.domain.common.Id

interface TodoListRepository {

    suspend fun getTodoListForUser(userId: Id): Result<TodoList?>

    suspend fun addTodoList(todoList: TodoList): Result<Unit>

    suspend fun updateTodoList(todoList: TodoList): Result<Unit>

}

File: ios/YATA/Domain/TodoList/TodoListRepository.swift
//
//  TodoListRepository.swift
//  YATA
//
//  Created by Alpár Szotyori on 26.05.24.
//

import Foundation

protocol TodoListRepository {
    func getTodoListForUser(_ userId: Id) async -> Result<TodoList?, Error>

    func addTodoList(_ todoList: TodoList) async -> Result<Void, Error>

    func updateTodoList(_ todoList: TodoList) async -> Result<Void, Error>
}
